### PR TITLE
[FIX] stock_account: Performance of _compute_stock_value

### LINKED ISSUE #32524: [11.0] Stock Valuation report takes 20 mins to run in large DB
Impacted versions:

11.0

Steps to reproduce:

On a large DB (10+ millions moves), go to Inventory > Reporting > Inventory Valuation

Current behavior:

Wait 20+ minutes to get the result.

Expected behavior:

Wait less than 5 min?

Pyflame graph:
[Inventory-valuation.zip](https://github.com/odoo/odoo/files/3055634/Inventory-valuation.zip)

Enterprise Ticket:
https://www.odoo.com/my/task/1961412

### PR DESCRIPTION
**Pull request by ndeodoo**
On the ticket 1961412, the custommer has more than 12 millions
of stock moves. Before this commit, the search request  does
not take too much time to execute but fetching the fields on the
recordset takes a lot of time due to the 'in' in the 'where clause'
of the associated SQL request.

This optimization permit to avoid this 'where clause' and decrease
the time to gather all the different fields. The time has been reduce
to less than one minutes where it takes approximately 20 minutes before.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

Fixes #32524

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock_account/models/product.py [hunk 1]
```diff
@@ -201,18 +201,29 @@ def _compute_stock_value(self):
...
-            for move in StockMove.search(domain).with_context(prefetch_fields=False):
-                product_values[move.product_id] += move.value
-                product_move_ids[move.product_id].append(move.id)
+            value_field_name = 'value'
         else:
-            domain = [('product_id', 'in', self.ids)] + StockMove._get_all_base_domain()
-            for move in StockMove.search(domain).with_context(prefetch_fields=False):
-                product_values[move.product_id] += move.remaining_value
-                product_move_ids[move.product_id].append(move.id)
+             domain = [('product_id', 'in', self.ids)] + StockMove._get_all_base_domain()
```
Comment (nim-odoo) [inline addons/stock_account/models/product.py]: wrong indentation

File: odoo/models.py [hunk 2]
```diff
@@ -4226,7 +4226,7 @@ def resolve_2many_commands(self, field_name, commands, fields=None):
 
     # for backward compatibility
     resolve_o2m_commands_to_record_dicts = resolve_2many_commands
-
+
```
Comment (nim-odoo) [inline odoo/models.py]: to remove

Comment (ndeodoo) [reply odoo/models.py]: Done

File: addons/stock_account/models/product.py (hunk 1 above)
Comment (ndeodoo) [reply addons/stock_account/models/product.py]: Done

**Comment by max3903** [comment]
https://github.com/odoo/odoo/issues/32524

**Comment by max3903** [comment]
:+1: It took less than a minute with this fix. Thank you!

File: addons/stock_account/models/product.py [hunk 3]
```diff
@@ -201,18 +201,29 @@ def _compute_stock_value(self):
...
-                product_values[move.product_id] += move.value
-                product_move_ids[move.product_id].append(move.id)
+            value_field_name = 'value'
         else:
             domain = [('product_id', 'in', self.ids)] + StockMove._get_all_base_domain()
-            for move in StockMove.search(domain).with_context(prefetch_fields=False):
-                product_values[move.product_id] += move.remaining_value
-                product_move_ids[move.product_id].append(move.id)
+            value_field_name = 'remaining_value'
+
```
[the remark this answers is not in the document]
Comment (ndeodoo) [reply addons/stock_account/models/product.py]: I add the check :)

**Comment by sle-odoo** [comment]
@nim-odoo looks good to me and the tests are green, maybe you want to have a last look before r+ing?

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**